### ISSUE TITLE
"E117: Unknown function: gtags#update" when I press "SPC m g c" to generate tags file.

### ISSUE THREAD

**Issue by shawnyangcis** [wsdjeg/SpaceVim#2097]
<!-- bug reporting without issue template will be closed automatically -->
When I pressed _SPC m g c_ or _SPC m g u_, and the error info in title was printed. There was no tags file in project dir. But I can make tags file using `:!gtags`, and other three options is avaliable.
## Expected behavior, english is required
Press SPC m g c, and then the gtags file will be generated in current project dir.
## The reproduce ways from Vim starting (Required!)
1. Open a project with nvim, ensure that no tags file existing.
2. Press _SPC m g c_ or _SPC m g u_, and the error message shows up.
## Debug info
 SPDebugInfo                                                                                                                                                                                  
<details><summary> SpaceVim debug information </summary>                                                                                                                                      
                                                                                                                                                                                              
### SpaceVim options :                                                                                                                                                                        
                                                                                                                                                                                              
```toml                                                                                                                                                                                       
  auto_disable_touchpad = 1                                                                                                                                                                   
  autocomplete_method = 'deoplete'                                                                                                                                                            
  autocomplete_parens = 1                                                                                                                                                                     
  background = 'dark'                                                                                                                                                                         
  buffer_index_type = 4                                                                                                                                                                       
  checkinstall = 1                                                                                                                                                                            
  colorscheme = 'nord'                                                                                                                                                                        
  colorscheme_bg = 'dark'                                                                                                                                                                     
  colorscheme_default = 'desert'                                                                                                                                                              
  custom_color_palette = []                                                                                                                                                                   
  custom_plugins = []                                                                                                                                                                         
  debug_level = 1                                                                                                                                                                             
  default_indent = 4                                                                                                                                                                          
  disabled_plugins = []                                                                                                                                                                       
  enable_ale = 0                                                                                                                                                                              
  enable_cursorcolumn = 0                                                                                                                                                                     
  enable_cursorline = 1                                                                                                                                                                       
  enable_debug = 0                                                                                                                                                                            
  enable_googlesuggest = 0                                                                                                                                                                    
  enable_guicolors = 1                                                                                                                                                                        
  enable_insert_leader = 1                                                                                                                                                                    
  enable_javacomplete2_py = 0                                                                                                                                                                 
  enable_key_frequency = 0                                                                                                                                                                    
  enable_language_specific_leader = 1                                                                                                                                                         
  enable_neocomplcache = 0                                                                                                                                                                    
  enable_neomake = 1                                                                                                                                                                          
  enable_os_fileformat_icon = 0                                                                                                                                                               
  enable_powerline_fonts = 1                                                                                                                                                                  
  enable_statusline_mode = 0                                                                                                                                                                  
  enable_tabline_filetype_icon = 1                                                                                                                                                            
  enable_vimfiler_filetypeicon = 0                                                                                                                                                            
  enable_vimfiler_gitstatus = 0                                                                                                                                                               
  enable_vimfiler_welcome = 1                                                                                                                                                                 
  enable_ycm = 0                                                                                                                                                                              
  error_symbol = '✖'                                                                                                                                                                          
  expand_tab = 1                                                                                                                                                                              
  filemanager = 'vimfiler'                                                                                                                                                                    
  filetype_icons = {}                                                                                                                                                                         
  force_global_config = 0                                                                                                                                                                     
  gitcommit_issue_icon = ''                                                                                                                                                                  
  gitcommit_pr_icon = ''                                                                                                                                                                     
  github_username = ''                                                                                                                                                                        
  guicolors = 1                                                                                                                                                                               
  guifont = ''                                                                                                                                                                                
  hiddenfileinfo = 1                                                                                                                                                                          
  hosts_url = 'https://raw.githubusercontent.com/racaljk/hosts/master/hosts'                                                                                                                  
  info_symbol = 'ⓘ' 
  keep_server_alive = 1                                                                                                                                                                       
  language = ''                                                                                                                                                                               
  leader_guide_default_group_name = ''                                                                                                                                                        
  leader_guide_flatten = 1                                                                                                                                                                    
  leader_guide_hspace = 5                                                                                                                                                                     
  leader_guide_max_size = 0                                                                                                                                                                   
  leader_guide_position = 'botright'                                                                                                                                                          
  leader_guide_run_map_on_popup = 1                                                                                                                                                           
  leader_guide_sort_horizontal = 0                                                                                                                                                            
  leader_guide_submode_mappings = {'<C-C>': 'win_close'}                                                                                                                                      
  leader_guide_vertical = 0                                                                                                                                                                   
  lint_on_save = 1                                                                                                                                                                            
  lint_on_the_fly = 0                                                                                                                                                                         
  max_column = 120                                                                                                                                                                            
  plugin_bundle_dir = '/home/yangyinqi/.cache/vimfiles/'                                                                                                                                      
  plugin_manager = 'dein'                                                                                                                                                                     
  plugin_manager_processes = 16                                                                                                                                                               
  plugin_name = 'unite-tag'                                                                                                                                                                   
  project_rooter_automatically = 1                                                                                                                                                            
  project_rooter_patterns = ['.git/', '_darcs/', '.hg/', '.bzr/', '.svn/', '.clang', 'pom.xml', 'package.json']                                                                               
  realtime_leader_guide = 1                                                                                                                                                                   
  relativenumber = 0                                                                                                                                                                          
  search_tools = ['rg', 'ag', 'pt', 'ack', 'grep']                                                                                                                                            
  sidebar_width = 30                                                                                                                                                                          
  simple_mode = 0                                                                                                                                                                             
  smartcloseignoreft = ['tagbar', 'vimfiler', 'SpaceVimRunner', 'SpaceVimREPL', 'SpaceVimQuickFix', 'HelpDescribe', 'VebuggerShell', 'VebuggerTerminal', 'SpaceVimTabsManager']               
  smartcloseignorewin = ['__Tagbar__', 'vimfiler:default']                                                                                                                                    
  snippet_engine = 'neosnippet'                                                                                                                                                               
  src_root = 'E:\sources\'                                                                                                                                                                    
  statusline_display_mode = 0                                                                                                                                                                 
  statusline_inactive_separator = 'arrow'                                                                                                                                                     
  statusline_left_sections = ['winnr', 'filename', 'major mode', 'syntax checking', 'minor mode lighters', 'vcs', 'hunks']                                                                    
  statusline_right_sections = ['fileformat', 'cursorpos', 'percentage']                                                                                                                       
  statusline_separator = 'arrow'                                                                                                                                                              
  statusline_unicode_symbols = 1                                                                                                                                                              
  terminal_cursor_shape = 2                                                                                                                                                                   
  version = '0.9.0-dev'                                                                                                                                                                       
  vim_help_language = 'en'                                                                                                                                                                    
  vimcompatible = 0                                                                                                                                                                           
  warning_symbol = '⚠'                                                                                                                                                                        
  wildignore = '*/tmp/*,*.so,*.swp,*.zip,*.class,tags,*.jpg,*.ttf,*.TTF,*.png,*/target/*,.git,.svn,.hg,.DS_Store,*.svg'                                                                       
  windows_index_type = 0                                                                                                                                                                      
  windows_leader = 's'                                                                                                                                                                        
  windows_smartclose = 'q'                                                                                                                                                                    
```                                                                                                                                                                                           
                                                                                                                                                                                              
                                                                                                                                                                                              
### SpaceVim layers :                                                                                                                                                                         
                                                                                                                                                                                              
```toml                                                                                                                                                                                       
[[layers]]                                                                                                                                                                                    
  name="autocomplete"                                                                                                                                                                         
  auto-completion-return-key-behavior='complete'
  auto-completion-tab-key-behavior='smart'                                                                                                                                                    
[[layers]]                                                                                                                                                                                    
  name="checkers"                                                                                                                                                                             
[[layers]]                                                                                                                                                                                    
  name="format"                                                                                                                                                                               
[[layers]]                                                                                                                                                                                    
  name="edit"                                                                                                                                                                                 
[[layers]]                                                                                                                                                                                    
  name="ui"                                                                                                                                                                                   
[[layers]]                                                                                                                                                                                    
  name="core"                                                                                                                                                                                 
[[layers]]                                                                                                                                                                                    
  name="core#banner"                                                                                                                                                                          
[[layers]]                                                                                                                                                                                    
  name="core#statusline"                                                                                                                                                                      
[[layers]]                                                                                                                                                                                    
  name="core#tabline"                                                                                                                                                                         
[[layers]]                                                                                                                                                                                    
  name="shell"                                                                                                                                                                                
  default_position='bottom'                                                                                                                                                                   
  default_height=30                                                                                                                                                                           
[[layers]]                                                                                                                                                                                    
  name="VersionControl"                                                                                                                                                                       
[[layers]]                                                                                                                                                                                    
  name="git"                                                                                                                                                                                  
[[layers]]                                                                                                                                                                                    
  name="github"                                                                                                                                                                               
[[layers]]                                                                                                                                                                                    
  name="denite"                                                                                                                                                                               
[[layers]]                                                                                                                                                                                    
  name="lang#sh"                                                                                                                                                                              
[[layers]]                                                                                                                                                                                    
  name="lang#c"                                                                                                                                                                               
  libclang_path='/usr/lib/libclang.so'                                                                                                                                                        
  clang_executable='/usr/bin/clang'                                                                                                                                                           
[[layers]]                                                                                                                                                                                    
  name="lang#dockerfile"                                                                                                                                                                      
[[layers]]                                                                                                                                                                                    
  name="lang#java"                                                                                                                                                                            
[[layers]]                                                                                                                                                                                    
  name="lang#javascript"                                                                                                                                                                      
[[layers]]                                                                                                                                                                                    
  name="lang#python"                                                                                                                                                                          
[[layers]]                                                                                                                                                                                    
  name="lang#vim"                                                                                                                                                                             
[[layers]]                                                                                                                                                                                    
  name="lang#markdown"                                                                                                                                                                        
[[layers]]                                                                                                                                                                                    
  name="lang#html"                                                                                                                                                                            
[[layers]]                                                                                                                                                                                    
  name="colorscheme"                                                                                                                                                                          
[[layers]]
  name="tags"                                                                                                                                                                                 
[[layers]]                                                                                                                                                                                    
  name="cscope"                                                                                                                                                                               
[[layers]]                                                                                                                                                                                    
  name="debug"                                                                                                                                                                                
```                                                                                                                                                                                           
                                                                                                                                                                                              
                                                                                                                                                                                              
### SpaceVim Health checking :                                                                                                                                                                
                                                                                                                                                                                              
SpaceVim clipboard support check report:                                                                                                                                                      
Checking +clipboard:                                                                                                                                                                          
      SUCCEED!                                                                                                                                                                                
SpaceVim environment check report:                                                                                                                                                            
Current progpath: nvim(/usr/bin/nvim)                                                                                                                                                         
version: 800                                                                                                                                                                                  
OS: linux                                                                                                                                                                                     
[shell, shellcmdflag, shellslash]: ['/usr/bin/zsh', '-c', 0]                                                                                                                                  
SpaceVim lua support check report:                                                                                                                                                            
Checking +lua:                                                                                                                                                                                
      Failed : Known issue, neovim do not support lua now.                                                                                                                                    
SpaceVim python support check report:                                                                                                                                                         
Checking +python3:                                                                                                                                                                            
      SUCCEED!                                                                                                                                                                                
Checking +python:                                                                                                                                                                             
      SUCCEED!                                                                                                                                                                                
                                                                                                                                                                                              
### SpaceVim runtime log :                                                                                                                                                                    
                                                                                                                                                                                              
```log                                                                                                                                                                                        
[ SpaceVim ] : logger file  does not exists, only log for current process will be shown!                                                                                                      
[ SpaceVim ] [17:07:24] [ Info ] Can not find project local config, start to loadding global config                                                                                           
[ SpaceVim ] [17:07:24] [ Info ] SpaceVim server startup at:/tmp/spacevim_nvim_server                                                                                                         
[ SpaceVim ] [17:07:24] [ Info ] Find project root('.git/',''):/home/yangyinqi/code/work/f04/sched_engine/.git/                                                                               
[ SpaceVim ] [17:07:24] [ Info ] change to root:/home/yangyinqi/code/work/f04/sched_engine                                                                                                    
[ SpaceVim ] [17:07:24] [ Info ] change to root:/home/yangyinqi/code/work/f04/sched_engine                                                                                                    
[ SpaceVim ] [17:07:27] [ Info ] Find project root('.git/','/home/yangyinqi/code/work/f04/sched_engine/resolveTime.py'):/home/yangyinqi/code/work/f04/sched_engine/.git/                      
[ SpaceVim ] [17:07:27] [ Info ] change to root:/home/yangyinqi/code/work/f04/sched_engine                                                                                                    
```                                                                                                                                                                                           
</details>
Please press <kbd>SPC h I</kbd>, debug info will be put into clipboard, then paste all content below.

## Screenshots

If you have any screenshots for this issue, please upload here. BTW you can use https://asciinema.org/ for recording video in terminal.

<!-- please remove the issue template when request for a feature -->

**Comment by wsdjeg** [comment]
make sure gtags.vim is installed, it is inlcuded in tags layer.

**Comment by yinflying** [comment]
Same Problem Here. (nvim)
```
:GtagsGenerate
E117: Unknown function: gtags#update
```
And I can't call gtags#update at all. However, layer "tags" has been installed.

**Comment by benlindsay** [comment]
I had exactly the same issue, where the exact same error mentioned in this issue title showed up when I pressed `SPC m g c`. I do have

```
[[layers]]
name = "tags"
```
 in my `init.toml` file. I tried making sure gtags was installed by adding

```
function! myspacevim#before() abort
...
  let g:spacevim_custom_plugins = [
        \ ['vim-scripts/gtags.vim'],
        \ ]
...
endfunction
```

Now, `SPC m g d` works as expected, but when I press `SPC m g c` I still get an error, but a different one: `E492: Not an editor command: GtagsGenerate!`. I can still do `!gtags` but it would be nice if `SPC m g c` worked too. Any thoughts of what's going on?
